File: examples/elixir/basic-server/test/support/basic_service_server_probe.ex
defmodule BasicServiceServerProbe do
  @moduledoc false

  @spec handle_get_type_closure(term(), BasicServiceTypes.GetTypeClosureInput.t(), term()) ::
          {:ok, BasicServiceTypes.GetTypeClosureOutput.t()} | {:error, term()}
  def handle_get_type_closure(_ctx, input, _meta) do
    {:ok,
     %BasicServiceTypes.GetTypeClosureOutput{
       basic_string: input.name,
       basic_boolean: input.verbose
     }}
  end
end

File: examples/erlang/basic-server/test/basic_service_server_probe.erl
-module(basic_service_server_probe).

-include("basic_service_types.hrl").

-export([handle_get_type_closure/3]).

%% Test double: partial handler module for router tests (not a behaviour implementor).
handle_get_type_closure(_Ctx, #get_type_closure_input{name = Name, verbose = Verbose}, _Meta) ->
    {ok, #get_type_closure_output{
        basic_string = Name,
        basic_boolean = Verbose
    }}.
